process-dir.js

```diff
@@ -147,12 +147,10 @@ async function replacePackage (dir, file, files, envTargets) {
   pkg.module = 'index.js'
   pkg['react-native'] = 'index.js'
 
-  if (
-    envTargets.includes(file) ||
-    envTargets.includes(file.replace(/\.js$/, '.browser.js'))
-  ) {
+  if (envTargets.includes(file)) {
     pkg.browser = {
-      './index.js': './index.browser.js'
+      development: './index.dev.js',
```

Comment:
webpack doesn’t support this syntax. `package.browser` should have only `file` → `file` map.

Reply:
This transformation will be done only if there is no `index.browser.js`, so webpack will use `index.js`. Will test that with integration test now.

Reply:
`pkg.browser.development` is invalid syntax, everyone will just ignore it.

What do you want to do by these changes? Maybe there is a better way to do it.

Reply:
I added the test, it fails in many bundlers in version 1.0.0, but passes now. If there is `index.browser.js` bundlers we will get 
`browser: './index.browser.js'`

Otherwise we process `index.js` to separate dev/prod files

Reply:
Why do we need `pkg.browser = { development, production }` for non-`index.browser.js` cawe if nobody understand `pkg.browser.production`?

Reply:
According to this issue https://github.com/ai/nanoid/issues/237, some bundlers should respect `{ development, production }`.

As to: "why do we need it for non-`index.browser.js`", I thought the main reason for this issue was to provide a modern version of a package that doesn't have `process.env.NODE_ENV` inside.

It was mentioned in the [initial task](https://cultofmartians.com/tasks/dual-publish-process-env.html#task) - that in absence of `index.browser.js` we should process `index.js`

I have an idea to try:
`{ development, production, default }`
maybe this will work for webpack

Reply:
> According to this issue ai/nanoid#237, some bundlers should respect { development, production }.

Bundlers support `pkg.exports.browser = { development, production }` not `pkg.browser = { development, production }`

Reply:
default: doesn't work, webpack only respects simple `'browser': './index.js'`

Reply:
Yeap, webpack prefers `pkg.browser` (DO NOT support environments) over `pkg.exports.browser` (supports environments).

But webpack anyway support `process.env.NODE_ENV`, so it is not a big deal.

Reply:
Mayb we need to add `esbuild` test to be sure that `pkg.exports.browser.production` was used

Reply:
If we can't process `index.browser.js` and processing `index.js` is useless, then [this issue](https://github.com/ai/nanoid/issues/237) can't be solved and we don't need processing at all.

Though it's stated [here](https://github.com/ai/nanoid/issues/237#issuecomment-703102113) in the second item that webpack should support development and production.

Maybe I miss something?